### DISCUSSION TITLE
How to convert a string to an angle?

### DISCUSSION THREAD

**Discussion by ttmc** [typst/typst#887] [Q&A] [answered]
How do I convert something of type [string](https://typst.app/docs/reference/types/string/), such as `"10deg"`, to something of type [angle](https://typst.app/docs/reference/types/angle/), such as `10deg`?

The `int()` and `float()` functions don't work (nor would I expect them to). There doesn't seem to be an `angle()` function. (I tried.)

**Comment by Enivex** [comment]
What is the use-case for this exactly?

**Comment by PgBiel** [comment, marked as answer]
You can use `eval("10deg")`, or parse the string by yourself with regex to get the number and multiply by 1deg.

**Comment by ttmc** [comment]
@Enivex I have a big array of floats such as `11.3`. They are angles in degrees. I'm able to convert them to strings such as `"11.3deg"` but I need angles because they're needed as values for `angle` in a call to `line(..., angle: <need an angle here>)`.

@PgBiel I used `eval()` as you suggested and it worked. Thanks, but surely that's a hack. (The docs for `eval()` say, "This function should only be used as a last resort.") I also tried `line(..., angle: 1deg * angle_as_float)` and that also worked, but it also feels like a hack.

**Reply by Enivex** [reply]
You don't actually have to go through a string at all. Just multiply the floats with 1deg, and you'll get angles.

Edit: I see you mention that. It's definitely not a hack.
